lib/ansible/executor/task_queue_manager.py

```diff
@@ -371,9 +372,20 @@ def send_callback(self, method_name, *args, **kwargs):
                 if gotit is not None:
                     methods.append(gotit)
 
+            # send clean copies
+            new_args = []
+            for arg in args:
+                # FIXME: add play/task cleaners
+                if isinstance(arg, TaskResult):
+                    new_args.append(arg.clean_copy())
```

Comment:
I suspect this might break more esoteric uses of callbacks (if something is relying on the original reference...).
But I think that is ok since that isnt really something we can (or should) guarantee.

Reply:
callbacks should not be manipulating the 'live objects', it can create issues and alter play flow in 'magical' ways the user would not understand/see/be able to audit. 

In any case we are moving this here to avoid doing it in the callback itself should not worry about security nor disclosures.